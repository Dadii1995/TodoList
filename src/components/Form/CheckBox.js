import React from 'react'
import { FormGroup, Label, Input } from 'reactstrap'

const CheckBox = ({ field: { name, ...field }, form: { touched, errors }, ...props }) => (
  <FormGroup check>
    <Label check>
      <Input data-cy={props.dataCy} type="checkbox" {...field} name={name} />{' '}
      {props.label && props.label}
    </Label>
  </FormGroup>
)

export default CheckBox
