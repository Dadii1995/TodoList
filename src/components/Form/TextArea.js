import React from 'react'
import { FormGroup, Label, Input, FormFeedback } from 'reactstrap'

const TextArea = ({ field: { name, ...field }, form: { touched, errors }, ...props }) => (
  <FormGroup>
    {props.label && <Label>{props.label}</Label>}
    <Input
      autoComplete="off"
      type="textarea"
      {...field}
      aria-label={props.label}
      data-cy={props.dataCy}
      name={name}
      placeholder={props.placeholder}
    />
    {errors[name] && <FormFeedback>{errors[name]}</FormFeedback>}
  </FormGroup>
)

export default TextArea
