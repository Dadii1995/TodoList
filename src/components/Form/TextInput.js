import React from 'react'
import { FormGroup, Label, Input, FormFeedback } from 'reactstrap'

const TextInput = ({ field: { name, ...field }, form: { touched, errors }, ...props }) => (
  <FormGroup>
    {props.label && <Label>{props.label}</Label>}
    <Input
      autoComplete="off"
      innerRef={props.innerRef}
      type="text"
      {...field}
      aria-label={props.label}
      data-cy={props.dataCy}
      name={name}
      placeholder={props.placeholder}
    />
    {errors[name] && <FormFeedback>{errors[name]}</FormFeedback>}
  </FormGroup>
)

export default TextInput
