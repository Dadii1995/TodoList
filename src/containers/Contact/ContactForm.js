import React from 'react'
import { Field, Form, Formik } from 'formik'
import { Alert, Button } from 'reactstrap'
import * as Yup from 'yup'

import TextInput from '../../components/Form/TextInput'
import SelectInput from '../../components/Form/SelectInput'
import TextArea from '../../components/Form/TextArea'
import CheckBox from '../../components/Form/CheckBox'

const formSchema = Yup.object().shape({
  email: Yup.string()
    .email('Invalid email')
    .required('Required'),
  content: Yup.string()
    .min(10, 'to short')
    .required('Required'),
  subject: Yup.string(),
  // accept: Yup.boolean()
  //   .oneOf([true], 'Must Accept Terms and Conditions')
  //   .required('Must Accept Terms and Conditions'),
})

const ContactForm = ({ onSubmit }) => {
  return (
    <Formik onSubmit={onSubmit} validationSchema={formSchema}>
      {({ isSubmitting, errors, handleSubmit }) => {
        return (
          <Form data-testid="test" onSubmit={handleSubmit}>
            <Field
              component={TextInput}
              label="E-Mail"
              name="email"
              placeholder="example@domain.com"
            />
            {errors.email ? <Alert color="danger">{errors.email}</Alert> : null}
            <Field
              component={SelectInput}
              label="Subject"
              name="subject"
              options={[
                { value: 'Technical Problems', label: 'Technical Problems' },
                { value: 'Orders', label: 'Orders' },
                { value: 'Other', label: 'Other' },
              ]}
            />
            {errors.subject ? <Alert color="danger">{errors.subject}</Alert> : null}
            <Field
              component={TextArea}
              label="Content"
              name="content"
              placeholder="Hi, blablalba"
            />
            {errors.content ? <Alert color="danger">{errors.content}</Alert> : null}
            <Field component={CheckBox} label="Accept terms" name="accept" />
            {errors.accept ? <Alert color="danger">{errors.accept}</Alert> : null}

            <Button color="warning" disabled={isSubmitting} type="submit">
              Send
            </Button>
          </Form>
        )
      }}
    </Formik>
  )
}
export default ContactForm
