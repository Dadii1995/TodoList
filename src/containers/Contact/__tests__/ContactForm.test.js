import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent, wait } from '@testing-library/react'

import ContactForm from '../ContactForm'
import TextInput from '../../../components/Form/TextInput'
import SelectInput from '../../../components/Form/SelectInput'
import TextArea from '../../../components/Form/TextArea'
import CheckBox from '../../../components/Form/CheckBox'

const createNodeMock = () => {
  const focus = () => console.log('Focused')
  return {
    focus,
  }
}

const getWrapper = () => {
  const wrapper = renderer.create(<ContactForm />, {createNodeMock})
  const root = wrapper.root
  return { root, wrapper }
}
describe('<ContactForm/>', () => {
  test('renders contact forms', () => {
    const { wrapper } = getWrapper()
    expect(wrapper).toMatchSnapshot()
  })
  test('renders text input', () => {
    const { root } = getWrapper()

    expect(root.findByProps({ component: TextInput })).toBeDefined()
  })
  test('renders select input', () => {
    const { root } = getWrapper()
    const select = root.findByProps({ component: SelectInput })

    expect(select).toBeDefined()
    expect(select.props.options).toHaveLength(3)
  })
  test('renders textarea', () => {
    const { root } = getWrapper()

    expect(root.findByProps({ component: TextArea })).toBeDefined()
  })
  test('renders checkbox', () => {
    const { root } = getWrapper()

    expect(root.findByProps({ component: CheckBox })).toBeDefined()
  })
  test('submit form', async () => {
    const onSubmit = jest.fn(x => x)
    const { getByText, getByLabelText } = render(<ContactForm onSubmit={onSubmit} />)

    const input = getByLabelText('E-Mail')
    fireEvent.change(input, { target: { value: 'aaa@aaa.aa' } })

    const select = getByLabelText('Subject')
    fireEvent.select(select, { target: { value: 'Orders' } })

    const textarea = getByLabelText('Content')
    fireEvent.change(textarea, { target: { value: 'test content 123456' } })

    const checkbox = getByLabelText('Accept terms')
    fireEvent.change(checkbox, { target: { value: true } })

    const button = getByText('Send')
    fireEvent.click(button)

    await wait(() => {
      expect(onSubmit).toHaveBeenCalled()
    })
  })
})
