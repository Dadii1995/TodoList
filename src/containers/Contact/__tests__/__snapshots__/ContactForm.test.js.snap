// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm/> renders contact forms 1`] = `
<form
  data-testid="test"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      E-Mail
    </label>
    <input
      aria-label="E-Mail"
      autoComplete="off"
      className="form-control"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="example@domain.com"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Subject
    </label>
    <select
      aria-label="Subject"
      className="form-control"
      name="subject"
      onBlur={[Function]}
      onChange={[Function]}
    >
      <option
        value="Technical Problems"
      >
        Technical Problems
      </option>
      <option
        value="Orders"
      >
        Orders
      </option>
      <option
        value="Other"
      >
        Other
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
    >
      Content
    </label>
    <textarea
      aria-label="Content"
      autoComplete="off"
      className="form-control"
      name="content"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Hi, blablalba"
    />
  </div>
  <div
    className="form-check"
  >
    <label
      className="form-check-label"
    >
      <input
        className="form-check-input"
        name="accept"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
      />
       
      Accept terms
    </label>
  </div>
  <button
    aria-label={null}
    className="btn btn-warning"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Send
  </button>
</form>
`;
